module/zfs/dsl_scan.c

```diff
@@ -611,11 +612,13 @@ dsl_scan_prefetch(dsl_scan_t *scn, arc_buf_t *buf, blkptr_t *bp,
 	    (BP_GET_LEVEL(bp) == 0 && BP_GET_TYPE(bp) != DMU_OT_DNODE))
 		return;
 
+	if (BP_IS_ENCRYPTED(bp))
+		zio_flags |= ZIO_FLAG_RAW;
```

Comment:
I did this so as not to interfere with existing functionality for unencrypted datasets. This line should only have an effect on encrypted dnodes, due to the check on line 612.